### PR TITLE
release docs: readability improvements

### PR DESCRIPTION
**Pull request by tstromberg**
#### Summary

Minor readability updates to the scaffolding release docs, based on following the instructions today for the v0.5.4 release.

#### Release Note

NONE

### CODE REVIEW

File: release.md [hunk 1]
```diff
@@ -7,51 +7,65 @@ You should be part of the `scaffolding-codeowners` or `sigstore-oncall` groups,
...
 
-### Once the `Release` action has been completed successfully, find your draft release at the [releases page](https://github.com/sigstore/scaffolding/releases)
+### Publish
 
-### Update the release notes by clicking on the `Generate release notes` button
+Once the `Release` action has been completed successfully:
 
-### Click the green "Publish release" button
+1. Find your draft release on the [releases page](https://github.com/sigstore/scaffolding/releases)
+2. Click the `Generate release notes` button to fill in the "What's Changed" section
```
Comment (vaikas) [inline release.md:69]: did you have to hit the 'pen' icon to get to this page (after L68)? I couldn't recall?

File: release.md [hunk 2]
```diff
@@ -7,51 +7,65 @@ You should be part of the `scaffolding-codeowners` or `sigstore-oncall` groups,
...
+3. Click the green `Publish release` button
+4. 🎉
```
Comment (vaikas) [inline release.md:71]: Did you have to check the 'make this the latest' or something checkbox before publish release above?

**Review by vaikas — COMMENTED** [review]
Looks great, just curious about small details that I couldn't recall if they were clear enough or did they just work?

**Review by vaikas — APPROVED** [review]
Thanks for doing the release and cleaning up the readme!

File: release.md (hunk 1 above)
Comment (tstromberg) [reply release.md:69]: I don't recall either - sorry!

File: release.md (hunk 2 above)
Comment (tstromberg) [reply release.md:71]: The right check boxes were pre-populated.

**Pull request merged**